File: frontend/src/types/session.ts
export const Gender = {
  Male: 0,
  Female: 1,
  Neutral: 2,
} as const

export type Gender = (typeof Gender)[keyof typeof Gender]

export const SessionStatus = {
  WaitingForPartner: 0,
  Active: 1,
  Completed: 2,
} as const

export type SessionStatus = (typeof SessionStatus)[keyof typeof SessionStatus]

export interface Session {
  id: string
  initiatorId: string
  partnerId: string | null
  targetGender: Gender
  joinCode: string
  partnerLink: string
  status: SessionStatus
  createdAt: string
  linkedAt: string | null
  isInitiator: boolean
  partnerDisplayName: string | null
  initiatorDisplayName: string | null
}

export interface CreateSessionRequest {
  targetGender: Gender
}

export interface JoinSessionRequest {
  joinCode: string
}

File: backend/NameMatch.Domain/Entities/Session.cs
using NameMatch.Domain.Enums;

namespace NameMatch.Domain.Entities;

public class Session
{
    public Guid Id { get; set; }
    public required string InitiatorId { get; set; }
    public string? PartnerId { get; set; }
    public Gender TargetGender { get; set; }
    public required string JoinCode { get; set; }
    public required string PartnerLink { get; set; }
    public SessionStatus Status { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime? LinkedAt { get; set; }

    // Preference tracking
    public SessionSetupStatus SetupStatus { get; set; }
    public DateTime? InitiatorPrefsCompletedAt { get; set; }
    public DateTime? PartnerPrefsCompletedAt { get; set; }

    // Filter tracking (hard filters run before soft preferences)
    public DateTime? InitiatorFiltersCompletedAt { get; set; }
    public DateTime? PartnerFiltersCompletedAt { get; set; }

    public ICollection<Vote> Votes { get; set; } = new List<Vote>();
    public ICollection<UserPreference> UserPreferences { get; set; } = new List<UserPreference>();
    public ICollection<SessionFilter> Filters { get; set; } = new List<SessionFilter>();
}
